platform/ui/src/components/DateRange/DateRange.jsx

```diff
@@ -0,0 +1,159 @@
+                isSelected ? 'PresetDateRangePicker_button__selected' : ''
+              }`}
+              onClick={() =>
+                onChange({
+                  startDate: start,
+                  endDate: end,
+                  preset: true,
+                })
+              }
+            >
+              {text}
+            </button>
+          );
+        })}
+      </div>
+    );
+  };
+  const renderMonthElement = ({ month, onMonthSelect, onYearSelect }) => {
+    const renderYearsOptions = () => {
+      const yearsRange = 20;
+      const options = [];
+
+      for (let i = 0; i < yearsRange; i++) {
+        const year = moment().year() - i;
+        options.push(
+          <option key={year} value={year}>
+            {year}
+          </option>
+        );
+      }
+
+      return options;
+    };
+
+    renderMonthElement.propTypes = {
+      onMonthSelect: PropTypes.func,
+      onYearSelect: PropTypes.func,
+    };
+
+    return (
+      <div className="flex justify-center">
+        <div className="my-0 mx-1">
+          <select
+            className="DateRangePicker_select"
+            value={month.month()}
+            onChange={e => onMonthSelect(month, e.target.value)}
+          >
+            {moment.months().map((label, value) => (
+              <option key={value} value={value}>
+                {label}
+              </option>
+            ))}
+          </select>
+        </div>
+        <div className="my-0 mx-1">
+          {}
+          <select
+            className="DateRangePicker_select"
+            value={month.year()}
+            onChange={e => onYearSelect(month, e.target.value)}
+          >
+            {renderYearsOptions()}
+          </select>
+        </div>
+      </div>
+    );
+  };
+
+  return (
+    <DateRangePicker
+      /** REQUIRED */
+      startDate={startDate}
+      startDateId={'startDateId'}
+      endDate={endDate}
+      endDateId={'endDateId'}
+      onDatesChange={onChange}
+      focusedInput={focusedInput}
+      onFocusChange={updatedVal => setFocusedInput(updatedVal)}
+      /** OPTIONAL */
+      renderCalendarInfo={renderDatePresets}
+      renderMonthElement={renderMonthElement}
+      startDatePlaceholderText={'Start Date'}
+      endDatePlaceholderText={'End Date'}
+      phrases={{
+        closeDatePicker: 'Close',
+        clearDates: 'Clear dates',
+      }}
+      isOutsideRange={day => !isInclusivelyBeforeDay(day, moment())}
+      hideKeyboardShortcutsPanel={true}
+      numberOfMonths={1}
+      showClearDates={false}
+      anchorDirection="left"
+    />
+  );
+};
+
+DateRange.defaultProps = {
+  startDate: null,
+  endDate: null,
+};
+
+DateRange.propTypes = {
+  /** Start date moment object */
+  startDate: PropTypes.object, // moment date is an object
+  /** End date moment object */
+  endDate: PropTypes.object, // moment date is an object
+  /** Callback that returns on object with selected dates */
+  onChange: PropTypes.func.isRequired,
```

Comment:
Maybe set all is required -- no defaults. Show example usage in MDX. Allows you to sidestep moment dependency. Set as devDependency.

Reply:
`react-dates` use Moment object as date values for `startDate` and `endDate`.

I set `startDate` and `endDate` as optional because their value can be null when the field is empty.

I'm using `moment` in our `DateRange` component to mount the custom Filter Presets (Today, Last 7 Days, Last 30 Days) and to render the customized month/year select. These fields are customized in our side, but I'm not exposing them as props of our `DateRange` to keep the wrapped component as simple as possible.

What do you think we can do here?